core/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -307,9 +330,26 @@ public boolean exists(Settings settings) {
      * instead.
      */
     public T get(Settings settings) {
+        return get(settings, true);
+    }
+
+    private T get(Settings settings, boolean validate) {
         String value = getRaw(settings);
         try {
-            return parser.apply(value);
+            T parsed = parser.apply(value);
+            if (validate) {
+                final Iterator<Setting<T>> it = validator.settings();
+                final Map<Setting<T>, T> map;
+                if (it.hasNext()) {
+                    map = new HashMap<>();
+                    final Setting<T> setting = it.next();
+                    map.put(setting, setting.get(settings, false)); // we have to disable validation or we will stack overflow
+                } else {
+                    map = Collections.emptyMap();
+                }
+                validator.validate(parsed, map);
```

Comment:
just a drive by question - why not pass the entire settings object and let the (exceptional) validate logic look at whatever they want?